plan/logical_plan_builder.go

```diff
@@ -1735,19 +1735,22 @@ func (b *planBuilder) buildDataSource(tn *ast.TableName) LogicalPlan {
 	schema := expression.NewSchema(make([]*expression.Column, 0, len(columns))...)
 	for i, col := range columns {
 		ds.Columns = append(ds.Columns, col.ToInfo())
-		schema.Append(&expression.Column{
+		newCol := &expression.Column{
 			FromID:   ds.id,
 			Position: i,
 			DBName:   dbName,
 			TblName:  tableInfo.Name,
 			ColName:  col.Name,
 			ID:       col.ID,
 			RetType:  &col.FieldType,
-		})
+		}
 
 		if tableInfo.PKIsHandle && mysql.HasPriKeyFlag(col.Flag) {
-			handleCol = schema.Columns[i]
+			newCol.RetType = newCol.RetType.Clone()
```

Comment:
how about:
```go
newCol.RetType = new(types.FieldType)
*newCol.RetType = *col. FieldType
```

Thus we no need to implement a `Clone`  function for `FieldType`.